### PR TITLE
Cleanup and lofi integration

### PR DESCRIPTION
**Pull request by oxapentane**
Draft for now

Stuff done:
- Local testing, no more shitting into our staging db
- improvements to logging
- SwaggerUI/OpenApi added back (plus some progress on #2)
- Added correlation endpoint (not all features are supported yet though, first need to migrate to new locations storage)

### CODE REVIEW

File: src/routes/run.rs [hunk 1]
```diff
@@ -32,14 +40,199 @@ pub struct SubmitTravel {
...
+        .first(&mut database_connection)
+    {
+        Ok(r) => r,
+        Err(e) => {
+            error!("While trying to query for run {}: {e}", corr_request.run_id);
+            return Err(ServerError::InternalError);
+        }
+    };
+
+    if run.owner != req_user.id && req_user.role != Role::Administrator as i32 {
```
Comment (revol-xut) [inline src/routes/run.rs:142]: Maybe smart to give stats for the owner and writing into storage exclusively for admins

File: src/routes/run.rs [hunk 2]
```diff
@@ -32,14 +40,199 @@ pub struct SubmitTravel {
...
+            return Err(ServerError::InternalError);
+        }
+    };
+
+    let queried_gps: Vec<InsertGpsPoint> = queried_gps.into_iter().map(|p| p.into()).collect();
+
+    // query r09 telegrams matching the timeframe of the run
+    use tlms::schema::r09_telegrams::dsl::r09_telegrams;
+    use tlms::schema::r09_telegrams::time as telegram_time;
+    let telegrams: Vec<R09SaveTelegram> = match r09_telegrams
```
Comment (revol-xut) [inline src/routes/run.rs:171]: maybe smart to add

```rust
	.filter(line.eq(run.line))
	.filter(run.eq(run.run))
```

File: src/routes/run.rs [hunk 3]
```diff
@@ -32,14 +40,199 @@ pub struct SubmitTravel {
...
+        }
+    };
+
+    // get region data cache
+    let cache_dir = env::var("TREKKIE_CACHE_DIR").unwrap_or("/tmp/trekkie".to_string());
+    let datacare_api =
+        env::var("TREKKIE_DATACARE_API").unwrap_or("https://datacare.dvb.solutions".to_string());
+    let corr_window: i64 = match env::var("TREKKIE_CORRELATE_WINDOW") {
+        Ok(w) => match w.parse() {
+            Ok(uwu) => uwu,
```
Comment (revol-xut) [inline src/routes/run.rs:192]: XD

**Review by revol-xut — CHANGES_REQUESTED** [review]

File: src/routes/run.rs (hunk 1 above)
Comment (oxapentane) [reply src/routes/run.rs:142]: I would shift this problem to future us, when we move stops.json into the db

File: src/routes/run.rs (hunk 3 above)
Comment (oxapentane) [reply src/routes/run.rs:196]: image

**Review by revol-xut — APPROVED** [review]

**Pull request merged**